### PR TITLE
Add Tableau Config to Web User Invitation

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

Adds header labels for sections and adds "Tableau Configuration" Section

Screen Shot 2024-05-23 at 12 54 33 PM

## Technical Summary

[Jira Ticket
](https://dimagi.atlassian.net/browse/USH-4452)

## Feature Flag

[Tableau User Syncing](https://www.commcarehq.org/hq/flags/edit/tableau_user_syncing/)

### Safety story

Locally Tested, changes limited to Web User invite workflows. 

### Automated test coverage

Automated tests exists that `add_web_user` correctly creates Web User with Tableau configs defined in the Invitation.

### QA Plan

[QA-6560](https://dimagi.atlassian.net/browse/QA-6560)

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
-  [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

[QA-6560]: https://dimagi.atlassian.net/browse/QA-6560?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: corehq/apps/users/views/__init__.py [hunk 1]
```diff
@@ -1164,6 +1172,9 @@ def post(self, request, *args, **kwargs):
                                          primary_location_id=data.get("primary_location", None),
                                          program_id=data.get("program", None),
                                          assigned_location_ids=data.get("assigned_locations", None),
+                                         profile=profile,
+                                         tableau_role=data.get("tableau_role", None),
+                                         tableau_group_ids=data.get("tableau_group_ids", None)
```
Comment (Robert-Costello) [inline corehq/apps/users/views/__init__.py:1177]: nice. I was going to ask you about this in https://github.com/dimagi/commcare-hq/pull/34618 but you've got it covered.

File: corehq/apps/registration/forms.py [hunk 2]
```diff
@@ -565,7 +588,26 @@ def clean_email(self):
 
     def clean(self):
         cleaned_data = super(AdminInvitesUserForm, self).clean()
+
+        if (('tableau_role' in cleaned_data or 'tableau_group_indices' in cleaned_data)
+        and self.can_edit_tableau_config):
```
Comment (Robert-Costello) [inline corehq/apps/registration/forms.py:593]: should this be a negative case, as in `and not and self.can_edit_tableau_config`?

Comment (Jtang-1) [reply corehq/apps/registration/forms.py]: yes! good catch thank you [c98633e](https://github.com/dimagi/commcare-hq/pull/34678/commits/c98633e223c61b4eaf5f4767ebe091e4c97e2912)

**Review by Robert-Costello — APPROVED** [review]
Comment on python3 approach to super isn't blocking.

**Review by AddisonDunn — COMMENTED** [review]
Haven't looked at it in full detail but it looks pretty good to me. I like the way you utilized the existing tableau form code in the invite form

**Review by nospame — APPROVED** [review]
Reviewed only the last commit by request, makes sense in context and looks like everything else was previously approved.

**Comment by Jtang-1** [comment]
Thanks for the review Robert and that last minute review request Evan!

**Pull request merged**